src/callbacks.jl

```diff
@@ -590,7 +590,10 @@ function bisection(f, tup, tdir ; maxiters=1000)
   fx0x1 = f(x0) * f(x1)
   fzero = zero(fx0x1)
   (fx0x1 >= fzero) && error("Non bracketing interval passed in bisection method. Please report the error in DiffEqBase.")
-  prevfloat_tdir(t,forward=isone(tdir)) = forward ? prevfloat(t) : nextfloat(t)
+  # The comment line below does not allow for inference, yet the second version does.
+  forward=isone(tdir)
+  #prevfloat_tdir = forward ? prevfloat : nextfloat
+  prevfloat_tdir(t) = forward ? prevfloat(t) : nextfloat(t)
```

Comment:
Haha I guess I'm not explaining well enough what I mean. I thought one doesn't have to use nested functions at all. Nevertheless, in this form maybe it would be inferrable even without defining `forward` at all?
I.e.
```suggestion
  prevfloat_tdir(t) = isone(tdir) ? prevfloat(t) : nextfloat(t)
```

Reply:
Oh yes it definitely infers in the suggestion you've made. I thought the concern was on the duplicated checks of isone(tdir) everytime prevfloat_tdir is called.

Are nested functions considered bad practice? I like the idea here as it is not something that another function should casually be using.

Reply:
No it's fine. And written in this way the check should only be performed once. It is just that if you would define a function `prevfloat_tdir(t, forward)` then one should precompute `isone(tdir)` instead of doing it over and over again in the loop.

Reply:
Okay, good stuff. I cleaned up the comments and left it inside and with the precomputed forward.